pkg/combustion/combustion.go

```diff
@@ -21,9 +22,44 @@ import (
 // Result can also be an empty slice or nil if this is not necessary.
 type configureComponent func(context *image.Context) ([]string, error)
 
+type networkConfigGenerator interface {
+	GenerateNetworkConfig(configDir, outputDir string, outputWriter io.Writer) error
+}
+
+type networkConfiguratorInstaller interface {
+	InstallConfigurator(sourcePath, installPath string) error
+}
+
+type kubernetesScriptDownloader interface {
+	DownloadInstallScript(distribution, destinationPath string) (string, error)
+}
+
+type kubernetesArtefactDownloader interface {
+	DownloadRKE2Artefacts(arch image.Arch, version, cni string, multusEnabled bool, installPath, imagesPath string) error
+	DownloadK3sArtefacts(arch image.Arch, version, installPath, imagesPath string) error
+}
+
+type rpmResolver interface {
+	Resolve(packages *image.Packages, localRPMConfig *image.LocalRPMConfig, outputDir string) (rpmDirPath string, pkgList []string, err error)
+}
+
+type rpmRepoCreator interface {
+	Create(path string) error
+}
+
+type Combustion struct {
```

Comment:
I went back and forth on this a few times. My initial idea in my head was that these component configuration objects would live under Builder, with it passing them into the combustion methods as necessary.

But now seeing this I kinda like the idea of all of these being encapsulated under the combustion object since, at least for now, they are SUPER scoped to generating combustion scripts and configuration. So +1 from me on this approach.